symfony/plugins/orangehrmLeavePlugin/Dto/LeaveRequest/DetailedLeave.php

```diff
@@ -0,0 +1,243 @@
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Leave\Dto\LeaveRequest;
+
+use OrangeHRM\Core\Traits\Auth\AuthUserTrait;
+use OrangeHRM\Entity\Leave;
+use OrangeHRM\Entity\WorkflowStateMachine;
+use OrangeHRM\Leave\Dto\LeaveDuration;
+use OrangeHRM\Leave\Dto\LeavePeriod;
+use OrangeHRM\Leave\Traits\Service\LeaveEntitlementServiceTrait;
+use OrangeHRM\Leave\Traits\Service\LeaveRequestServiceTrait;
+
+class DetailedLeave
+{
+    use LeaveEntitlementServiceTrait;
+    use LeaveRequestServiceTrait;
+    use AuthUserTrait;
+
+    /**
+     * @var Leave
+     */
+    private Leave $leave;
+
+    /**
+     * @var Leave|null
+     */
+    private ?Leave $firstLeaveOfLeavePeriod = null;
+
+    /**
+     * @var Leave[]
+     */
+    private array $leaves;
+
+    /**
+     * @var LeaveRequestDatesDetail|null
+     */
+    private ?LeaveRequestDatesDetail $dateDetail = null;
+
+    /**
+     * @var LeaveStatusWithLengthDays|null
+     */
+    private ?LeaveStatusWithLengthDays $status = null;
+
+    /**
+     * @var LeavePeriod|null
+     */
+    private ?LeavePeriod $leavePeriod = null;
+
+    /**
+     * @var LeaveBalanceWithLeavePeriod|null
+     */
+    private ?LeaveBalanceWithLeavePeriod $leaveBalance = null;
+
+    /**
+     * @var string[]|null
+     */
+    private ?array $allowedActions = null;
+
+    private function resetCache(): void
+    {
+        $this->dateDetail = null;
+        $this->leavePeriod = null;
+        $this->status = null;
+        $this->leaveBalance = null;
+        $this->allowedActions = null;
+    }
+
+    /**
+     * @return Leave
+     */
+    public function getLeave(): Leave
+    {
+        return $this->leave;
+    }
+
+    /**
+     * @param Leave $leave
+     */
+    public function setLeave(Leave $leave): void
+    {
+        $this->resetCache();
+        $this->leave = $leave;
+    }
+
+    /**
+     * @param Leave[] $leaves
+     */
+    public function setLeaves(array $leaves): void
+    {
+        $this->leaves = $leaves;
+    }
+
+    /**
+     * @return Leave[]
+     */
+    public function getLeaves(): array
```

Comment:
I'm not sure about this

Reply:
Used this function when getting the first leave of the leave period. Otherwise need to load them from DB every time when DetailedLeave is created.